### ISSUE TITLE
update Eigen3 

### ISSUE THREAD

**Issue by prudhomm** [feelpp/feelpp#828]
Eigen 3 should be upgraded to latest developments from bitbucket

**Comment by prudhomm** [comment]
here is the result of running fsi_2d 
````
finish-step1 in 4.00003e-06
finish-step2 in 1.00024e-06
finish-step3 in 0.000113
 it 2  nbLocalisationFail 0

||==============================================||
||==============================================||
||==============================================||
||----------Info : FluidMechanics---------------||
||==============================================||
||==============================================||
||==============================================||
   Prefix : fluid
   Root Repository : /home/prudhomm/feel/applications/models/fsi/TurekHron/fsi3/P2P1G1-P1G1/np_4
   Physical Model
     -- pde name  : Navier-Stokes
     -- stress tensor law  : newtonian
     -- time mode : Transient
     -- ale mode  : Yes
     -- gravity  : false
   Physical Parameters
     -- rho : 1050
     -- mu  : 0.00345
     -- nu  : 3.28571e-06
   Boundary conditions
       -- Dirichlet (elimination) : fluid-cylinder , fluid-inlet , fluid-wall
       -- fluid outlets [free] : fluid-outlet
       -- MeshALE[fixed] : fluid-cylinder , fluid-inlet , fluid-wall , fluid-outlet
       -- MeshALE[moving] : fsi-wall
   Space Discretization
     -- mesh file name   : /home/prudhomm/feel/applications/models/fsi/TurekHron/fsi3/P2P1G1-P1G1/meshes/fsi_fluid_Simplex_2_1_2_M1_p4.json
     -- nb elt in mesh  : 2654
     -- geometry order  : 1
     -- velocity order  : 2
     -- pressure order  : 1
     -- nb dof (u+p)    : 12498 (3179)
     -- nb dof (u)      : 11060
     -- nb dof (p)      : 1438
     -- stabilisation   : OFF
   Time Discretization
     -- initial time : 0
     -- final time   : 10
     -- time step    : 0.005
     -- order        : 2
     -- restart mode : No
     -- save on disk : true
     -- freq save : 1
     -- file format save : binary
   Exporter
     -- type            : ensightgold
     -- high order visu : OFF
     -- freq save       : 1
     -- fields exported : velocity - pressure - displacement - pid
   Processors
     -- number of proc environnement : 4
     -- environement rank : 0
     -- global rank : 0
     -- local rank : 0
   Numerical Solver
     -- solver : Oseen
||==============================================||
||---------------Info : SolverNum---------------||
||==============================================||
   Backend 
     -- prefix : fluid
     -- type : BACKEND_PETSC
   KSP 
     -- type : gmres
     -- tolerance : res=1e-08, diff=100000, abs=1e-50
     -- maxit : 1000
     -- maxit : 1000
     -- reuse prec : true
   SNES 
     -- reuse jac : true
     -- reuse jac rebuild at first Newton step : true
     -- tolerance : res=1e-08, step=1e-08, abs=1e-50
     -- maxit : 50
     -- maxit with reuse : 50
   KSP in SNES 
     -- tolerance : res=1e-05
     -- maxit : 1000
     -- maxit with reuse : 1000
     -- reuse prec : true
     -- reuse prec rebuild at first Newton step : true
   PC 
     -- type : lu
     -- mat solver package : mumps
||==============================================||
||----------------Info : MeshALE----------------||
||==============================================||
   Prefix : fluid.alemesh
   Physical Markers
     -- fixed : 4 , 2 , 3 , 5
     -- moving : 6
     -- free : 
   Harmonic Extension 
     -- type : yes
||==============================================||

||==============================================||
||==============================================||
||==============================================||

||==============================================||
||----------Info : SolidMechanics---------------||
||==============================================||
   Prefix : solid
   Root Repository : /home/prudhomm/feel/applications/models/fsi/TurekHron/fsi3/P2P1G1-P1G1/np_4
   Physical Model
     -- pde name : Hyper-Elasticity
     -- material law : StVenantKirchhoff
     -- use displacement-pressure formulation : false
     -- time mode : Transient
   Materials parameters
     -- number of materials : 1
     -- [Solid] rho : 1000
     -- [Solid] young modulus : 5.6e+06
     -- [Solid] coeff poisson : 0.4
     -- [Solid] coeff Lame 1 : 8e+06
     -- [Solid] coeff Lame 2 : 2e+06
   Boundary conditions
       -- Dirichlet (elimination) : solid-fixed
   Space Discretization
     -- msh file name   : /home/prudhomm/feel/applications/models/fsi/TurekHron/fsi3/P2P1G1-P1G1/meshes/fsi_solid_Simplex_2_1_2_M1_p4.json
     -- nb elt in mesh : 155
     -- nb dof (displacement) : 234
     -- polynomial order : 1
     -- mechanical properties : P0d
   Time Discretization
     -- initial time : 0
     -- final time   : 10
     -- time step    : 0.005
     -- type : Newmark ( gamma=0.5, beta=0.25 )
     -- restart mode : No
     -- save on disk : true
     -- freq save : 1
     -- file format save : binary
   Exporter
     -- type            : ensightgold
     -- high order visu : OFF
     -- freq save       : 1
     -- fields exported : displacement - velocity - pid
   Processors
     -- number of proc environnement : 4
     -- environement rank : 0
     -- global rank : 0
     -- local rank : 0
   Numerical Solver
     -- solver : Newton
||==============================================||
||---------------Info : SolverNum---------------||
||==============================================||
   Backend 
     -- prefix : solid
     -- type : BACKEND_PETSC
   KSP 
     -- type : gmres
     -- tolerance : res=1e-08, diff=100000, abs=1e-50
     -- maxit : 1000
     -- maxit : 1000
     -- reuse prec : true
   SNES 
     -- reuse jac : false
     -- reuse jac rebuild at first Newton step : true
     -- tolerance : res=1e-08, step=1e-08, abs=1e-50
     -- maxit : 50
     -- maxit with reuse : 10
   KSP in SNES 
     -- tolerance : res=1e-05
     -- maxit : 1000
     -- maxit with reuse : 10
     -- reuse prec : true
     -- reuse prec rebuild at first Newton step : true
   PC 
     -- type : lu
     -- mat solver package : mumps
||==============================================||

||==============================================||
||-----------------Info : FSI-------------------||
||==============================================||
   Coupling type : Semi-Implicit
   Coupling BC : dirichlet-neumann
   Interface property : conformal
   Solver
     -- method        : fix point with Aitken relaxation
     -- tolerance     : 1e-06
     -- maxit         : 1000
     -- initial theta : 0.98
     -- min theta     : 1e-12
||==============================================||

=====================================================================================
 current time : 0.00500
=====================================================================================
[Aitken] iteration : 1 theta=9.80000e-01 residualNorm : 1.00000e+00
[Aitken] iteration : 2 theta=6.18552e-01 residualNorm : 6.21126e-01
[Aitken] iteration : 3 theta=6.28110e-02 residualNorm : 5.03165e+00
[Aitken] iteration : 4 theta=5.38795e-02 residualNorm : 8.73495e-01
[Aitken] iteration : 5 theta=5.40609e-02 residualNorm : 8.11003e-02
[Aitken] iteration : 6 theta=4.99112e-01 residualNorm : 7.44121e-02
[Aitken] iteration : 7 theta=5.60769e-01 residualNorm : 2.89065e-02
[Aitken] iteration : 8 theta=2.11088e-01 residualNorm : 4.43660e-02
[Aitken] iteration : 9 theta=1.79005e-01 residualNorm : 8.28833e-03
[Aitken] iteration : 10 theta=1.82736e-01 residualNorm : 1.58862e-03
[Aitken] iteration : 11 theta=7.58257e-02 residualNorm : 1.83651e-03
[Aitken] iteration : 12 theta=5.61640e-02 residualNorm : 1.04903e-03
[Aitken] iteration : 13 theta=6.19017e-02 residualNorm : 7.61183e-04
[Aitken] iteration : 14 theta=4.74271e-01 residualNorm : 6.73287e-04
[Aitken] iteration : 15 theta=5.17090e-01 residualNorm : 1.34097e-04
[Aitken] iteration : 16 theta=1.11515e-01 residualNorm : 2.78487e-04
[Aitken] iteration : 17 theta=5.39978e-02 residualNorm : 2.94555e-04
[Aitken] iteration : 18 theta=5.42507e-02 residualNorm : 8.80411e-05
[Aitken] iteration : 19 theta=9.80000e-01 residualNorm : 8.45955e-05
[Aitken] iteration : 20 theta=9.80000e-01 residualNorm : 3.35076e-05
[Aitken] iteration : 21 theta=5.03260e-01 residualNorm : 3.19057e-05
[Aitken] iteration : 22 theta=3.35751e-01 residualNorm : 1.80739e-05
[Aitken] iteration : 23 theta=8.44682e-02 residualNorm : 4.03492e-05
[Aitken] iteration : 24 theta=5.40587e-02 residualNorm : 2.27831e-05
[Aitken] iteration : 25 theta=5.38083e-02 residualNorm : 2.01915e-06
[Aitken] iteration : 26 theta=4.24881e-01 residualNorm : 1.83881e-06
[Aitken] iteration : 27 theta=4.24881e-01 residualNorm : 7.99330e-07
[fsi.FSI] finish fsi solve in 6.73284e+00

=====================================================================================
 current time : 0.01000
=====================================================================================
[Aitken] iteration : 1 theta=9.80000e-01 residualNorm : 7.99330e-07
[Aitken] iteration : 2 theta=9.80000e-01 residualNorm : 1.92670e+00
[Aitken] iteration : 3 theta=1.09451e-01 residualNorm : 2.93100e+00
[Aitken] iteration : 4 theta=5.39395e-02 residualNorm : 3.12395e+00
[Aitken] iteration : 5 theta=5.39510e-02 residualNorm : 2.84556e-01
[Aitken] iteration : 6 theta=6.97795e-01 residualNorm : 2.66840e-01
[Aitken] iteration : 7 theta=7.03633e-01 residualNorm : 3.31389e-03
[Aitken] iteration : 8 theta=3.49195e-01 residualNorm : 3.35981e-03
[Aitken] iteration : 9 theta=1.65107e-01 residualNorm : 3.66896e-03
[Aitken] iteration : 10 theta=7.83683e-02 residualNorm : 3.95204e-03
[Aitken] iteration : 11 theta=5.38439e-02 residualNorm : 1.92996e-03
[Aitken] iteration : 12 theta=5.47910e-02 residualNorm : 7.07143e-04
[Aitken] iteration : 13 theta=7.02837e-01 residualNorm : 6.62801e-04
[Aitken] iteration : 14 theta=7.86841e-01 residualNorm : 2.36711e-04
[Aitken] iteration : 15 theta=2.00225e-01 residualNorm : 6.02748e-04
[Aitken] iteration : 16 theta=1.78948e-01 residualNorm : 7.50368e-05
[Aitken] iteration : 17 theta=1.46899e-01 residualNorm : 3.50647e-05
[Aitken] iteration : 18 theta=5.40700e-02 residualNorm : 5.78118e-05
[Aitken] iteration : 19 theta=5.41474e-02 residualNorm : 9.88662e-06
[Aitken] iteration : 20 theta=7.54937e-01 residualNorm : 9.26619e-06
[Aitken] iteration : 21 theta=8.18156e-01 residualNorm : 2.31642e-06
[Aitken] iteration : 22 theta=3.57677e-01 residualNorm : 2.88661e-06
[Aitken] iteration : 23 theta=3.57677e-01 residualNorm : 9.18958e-07
[fsi.FSI] finish fsi solve in 6.68998e+00

=====================================================================================
 current time : 0.01500
=====================================================================================
[Aitken] iteration : 1 theta=9.80000e-01 residualNorm : 9.18958e-07
[Aitken] iteration : 2 theta=1.52475e-01 residualNorm : 4.65570e+00
[Aitken] iteration : 3 theta=4.54526e-01 residualNorm : 2.19105e+00
[Aitken] iteration : 4 theta=1.96656e-01 residualNorm : 1.22419e+00
[Aitken] iteration : 5 theta=5.43673e-02 residualNorm : 2.79087e+00
[Aitken] iteration : 6 theta=5.39506e-02 residualNorm : 2.71482e-01
[Aitken] iteration : 7 theta=3.31277e-01 residualNorm : 2.47793e-01
[Aitken] iteration : 8 theta=7.65662e-01 residualNorm : 1.29942e-01
[Aitken] iteration : 9 theta=7.71686e-01 residualNorm : 2.24172e-03
[Aitken] iteration : 10 theta=5.80708e-02 residualNorm : 1.61134e-02
[Aitken] iteration : 11 theta=5.38686e-02 residualNorm : 1.77349e-03
[Aitken] iteration : 12 theta=5.77948e-02 residualNorm : 1.16148e-03
[Aitken] iteration : 13 theta=6.72107e-01 residualNorm : 1.07889e-03
[Aitken] iteration : 14 theta=7.45298e-01 residualNorm : 3.59322e-04
[Aitken] iteration : 15 theta=1.99808e-01 residualNorm : 8.62033e-04
[Aitken] iteration : 16 theta=1.79727e-01 residualNorm : 1.03696e-04
[Aitken] iteration : 17 theta=1.37193e-01 residualNorm : 5.83327e-05
[Aitken] iteration : 18 theta=5.47230e-02 residualNorm : 8.32824e-05
[Aitken] iteration : 19 theta=5.41045e-02 residualNorm : 2.13419e-05
[Aitken] iteration : 20 theta=4.95296e-01 residualNorm : 2.00783e-05
[Aitken] iteration : 21 theta=9.38026e-01 residualNorm : 9.51955e-06
[Aitken] iteration : 22 theta=9.38026e-01 residualNorm : 8.85204e-07
[fsi.FSI] finish fsi solve in 5.82893e+00
```

**Comment by prudhomm** [comment]
travis seems to be ok
https://travis-ci.org/feelpp/feelpp/builds/187141584

**Comment by prudhomm** [comment]
here are the results of the testsuite
```
94% tests passed, 20 tests failed out of 331

Label Time Summary:
testalg              =  14.79 sec (22 tests)
testcore             =  12.64 sec (36 tests)
testcrb              =  21.41 sec (9 tests)
testdiscr            = 828.16 sec (94 tests)
testfilters          = 320.40 sec (10 tests)
testfit              =   1.41 sec (4 tests)
testintegration      =  15.07 sec (14 tests)
testinterpolation    =  76.64 sec (25 tests)
testleaks            =  11.11 sec (16 tests)
testls               =   1.50 sec (3 tests)
testmaterial         =   0.49 sec (2 tests)
testmath             =   0.40 sec (2 tests)
testmesh             =  31.75 sec (24 tests)
testmodels           =   1.54 sec (2 tests)
testopt              =   0.60 sec (2 tests)
testpde              =  31.54 sec (4 tests)
testpoly             =  25.57 sec (18 tests)
testts               =   3.18 sec (5 tests)
testvf               = 696.94 sec (38 tests)

Total Test time (real) = 1076.73 sec

The following tests FAILED:
	 48 - feelpp_test_vector-np-1 (Failed)
	104 - feelpp_test_importergmsh-np-6 (Timeout)
	210 - feelpp_test_projector-np-6 (Timeout)
	211 - feelpp_test_projector-np-1 (Failed)
	217 - feelpp_test_submesh-np-1 (Failed)
	249 - feelpp_test_element0D-np-6 (Failed)
	250 - feelpp_test_element0D-np-1 (Failed)
	258 - feelpp_test_db-np-6 (Failed)
	262 - feelpp_test_bdf2-np-6 (Failed)
	263 - feelpp_test_bdf2-np-1 (Failed)
	264 - feelpp_test_bdf3-np-1 (Failed)
	271 - feelpp_test_laplacianv-np-6 (Failed)
	272 - feelpp_test_laplacianv-np-1 (Failed)
	273 - feelpp_test_laplacian-np-6 (Timeout)
	274 - feelpp_test_laplacian-np-1 (Failed)
	275 - feelpp_test_laplaciant-np-6 (Failed)
	276 - feelpp_test_laplaciant-np-1 (Failed)
	287 - feelpp_test_convolve-np-6 (Timeout)
	328 - feelpp_test_nofeel_interpolator-np-6 (Failed)
	329 - feelpp_test_nofeel_interpolator-np-1 (Failed)
Errors while running CTest
```

**Comment by prudhomm** [comment]
fixes #828
